pkg/fileutils/fileutils_test.go

```diff
@@ -624,3 +626,58 @@ func TestMatchesAmount(t *testing.T) {
 		assert.Equal(t, testCase.isMatch, isMatch, desc)
 	}
 }
+
+func TestExist(t *testing.T) {
+	tempDir := t.TempDir()
+
+	symlinkPath := path.Join(tempDir, "sl-working")
+	danglingSymlinkPath := path.Join(tempDir, "sl-broken")
+
+	err := os.Symlink(tempDir, symlinkPath)
+	require.NoError(t, err)
+
+	err = os.Symlink("fooobar123", danglingSymlinkPath)
+	require.NoError(t, err)
+
+	var syscallErr *syscall.Errno
+
+	err = Lexist(tempDir)
+	_, err2 := os.Stat(tempDir)
+	assert.Equal(t, errors.As(err, &syscallErr), errors.As(err2, &syscallErr), "same error for existing path")
+
+	err = Lexist("foo123shouldnotexist")
+	_, err2 = os.Stat("foo123shouldnotexist")
+	assert.Equal(t, errors.As(err, &syscallErr), errors.As(err2, &syscallErr), "same error for not existing path")
+
+	err = Lexist(symlinkPath)
+	_, err2 = os.Stat(symlinkPath)
+	assert.Equal(t, errors.As(err, &syscallErr), errors.As(err2, &syscallErr), "same error for existing symlink")
+
+	err = Exist(symlinkPath)
+	_, err2 = os.Lstat(symlinkPath)
+	assert.Equal(t, errors.As(err, &syscallErr), errors.As(err2, &syscallErr), "same error for existing symlink (follow=true)")
+
+	err = Lexist(danglingSymlinkPath)
+	_, err2 = os.Stat(danglingSymlinkPath)
+	assert.Equal(t, errors.As(err, &syscallErr), errors.As(err2, &syscallErr), "same error for not existing symlink")
+
+	err = Exist(danglingSymlinkPath)
+	_, err2 = os.Lstat(danglingSymlinkPath)
+	assert.Equal(t, errors.As(err, &syscallErr), errors.As(err2, &syscallErr), "same error for not existing symlink (follow=true)")
```

Comment:
- the `assert.Equal` is just comparing the bool returned form `errors.As`, i.e. wether the errors are of the same _type_, not that the values are the same. Is it intentional that this is what the test is checking?
- (The comments refer to `follow=true`, I suppose that should be updated to the new function names instead.)

Reply:
thanks, fixed and applied to the last version